src/utilities/checkMissingContext.ts

```diff
@@ -0,0 +1,19 @@
+import { MissingRouteContextError } from '@/errors/missingRouteContextError'
+import { Route, Routes } from '@/types/route'
+import { RouteContext } from '@/types/routeContext'
+
+export function checkMissingContext(routes: Routes): void {
+  const routeIdsFromContext = routes
+    .flatMap((route) => route.context)
+    .filter(contextIsRoute)
+
+  for (const route of routeIdsFromContext) {
+    if (routes.every(({ id }) => route.id !== id)) {
+      throw new MissingRouteContextError(route.name)
+    }
+  }
+}
```

Comment:
We should probably make this more efficient. It looks like we're looping over all the routes for each context route. Lets use a `Set` for all the route ids

```suggestion
export function checkMissingContext(routes: Routes): void {
  const routeIds = new Set(routes.map(route => route.id)
  const routeIdsFromContext = routes
    .flatMap((route) => route.context)
    .filter(contextIsRoute)

  for (const route of routeIdsFromContext) {
    if (!routeIds.has(route.id) {
      throw new MissingRouteContextError(route.name)
    }
  }
}
```